Update CODE_OWNERS.txt

jrose-apple opened the pull request:
On a recent PR, a new contributor wasn't sure who they should request as a reviewer; on another, I complained post-merge about the proper code owner not being notified. (Commit privileges doesn't mean you should be the final reviewer for *any* components, only those where it's clear the code owner trusts you.)

We don't advertise this file very much, and it doesn't fully solve the issue, but I think it's worth updating. This at least covers a few more of the cases that were slipping through the cracks.

jrose-apple committed:
CODE_OWNERS: Add GitHub accounts, for PR tagging purposes

jrose-apple committed:
CODE_OWNERS: Add a few uncontroversial unassigned components.

- Argyrios owns Index
- Jordan (me) owns Frontend, mostly as an extension of Driver
- Anna owns the sanitizers (ASan, etc.)

jrose-apple committed:
CODE_OWNERS: Add Adrian as the owner of debug info.

(as he has been for a long time)

jrose-apple committed:
CODE_OWNERS: Assign Max to all other overlays

The Apple framework overlays are still owned by their respective teams
at Apple, but Max is the person to contact if you need to make a
change there. Actual /additions/ should still be considered feature
requests that should go to Apple's bug tracker (bugreport.apple.com).

jrose-apple commented:
Any objections from @akyrtzi, @AnnaZaks, @adrian-prantl, or @moiseev?

For the owners of the "corelibs overlays", did I pick the right person and are you okay with being in this list? @parkera, @das / @mwwa, @mike-ferris-apple / @briancroom?

gparker42 commented:
Is Chris the right "everything else" owner?

briancroom commented:
I believe @mike-ferris-apple was aiming to reduce his ownership and I should rather be listed as the XCTest code owner.

adrian-prantl commented:
LGTM from my end.

jrose-apple commented:
@gparker42 I didn't want to coalesce that into this series of updates. These should be just writing down things that are true in practice, not changing or removing existing ownership (except from Chris's catch-all). @lattner and @tkremenek can work out the appropriate thing to do there.

tkremenek commented:
@lattner and I can discuss that catch-all separately, but I think it makes sense to be somebody else.  Let's tackle that one separately.  This LGTM.

AnnaZaks commented:
LGTM as well.

jrose-apple commented:
Updated to include @briancroom – would you prefer your Apple address or another address? (This is what you'd expect people to CC when they email the public lists.)

Still waiting on sign-off from: @akyrtzi, @parkera (or @phausler), and @das.

jrose-apple committed:
CODE_OWNERS: Add owners for the corelibs overlays.

- Tony P. owns the Foundation overlay.
- Daniel S. owns the Dispatch overlay.
- Brian C. owns the XCTest overlay.

(A reminder that ownership of the component does not mean they are the
only qualified reviewer for that component, nor does it necessarily
mean they are the head of the corresponding team at Apple)

akyrtzi commented:
LGTM

das commented:
lgtm

briancroom commented:
The Apple address is good. Thanks @jrose-apple!

jrose-apple commented:
@parkera / @phausler ping

CodaFi commented:
@jrose-apple In light of [Github's new Code Owners Feature](https://github.com/blog/2392-introducing-code-owners), can we co-opt this pull request to also move to the new format?

jrose-apple commented:
I'd want to check that with @tkremenek as project lead; unfortunately he's on vacation.

tkremenek commented:
@CodaFi @jrose-apple Let's make whether or not we use the new code owners feature a separate discussion on swift-dev first, and discuss how the feature should be used.  The real advantage of the new feature is how it can be used to gate pull requests being merged — which is something that should be discussed.  I'm happy to merge this pull request as is.

jrose-apple commented:
Rebased, but no changes from what I originally had laid out here.

@swift-ci Please smoke test
